### DISCUSSION TITLE
$embed_file should support string variable as well, not just string literals

### DISCUSSION THREAD

**Discussion by jdonnerstag** [vlang/v#11911] [Ideas]
**V version:**   V 0.2.2 1588fe7.be7c396
**OS:** windows, Microsoft Windows 10 Enterprise v19041 64-bit

**What did you do?**
This works fine
```
	mut file := $embed_file("rosie_unittest.rpl")
```
And this one compiles with an error
```
	fpath := "rosie_unittest.rpl"
	mut file := $embed_file(fpath)
```

**What did you expect to see?**
That $embed_file() not supports string literals as file name, but also string variables
 
**What did you see instead?**
```
error: unexpected name `fpath`, expecting string
  102 | fn load_rpl_file(debug int) ? rt.Rplx {
  103 |     fpath := "rosie_unittest.rpl"
  104 |     mut file := $embed_file(fpath)
      |                             ~~~~~
  105 |     data := file.to_string()
  106 |     mut p := parser.new_parser(data: data, fpath: fpath, debug: debug)?
```

**Comment by crthpl** [comment]
File embedding happens at compile time, and there is no way to know the value of a variable at compile time.

**Comment by jdonnerstag** [comment]
The variable is constant (not mutable) and a single literal (no computations), which I guess means that it is available at compile time.

**Comment by JalonSolov** [comment]
The variable still has to have space allocated at runtime, then a pointer to the string literal stored in it, before it looks like a "constant".

**Comment by spytheman** [comment]
It can be done in some limited cases (where the compiler knows that the value of the 'variable' is a string literal for sure), but not universally. V just does not do that analysis yet.

**Comment by jdonnerstag** [comment]
I was thinking about why I used $embed_file in the first place. I'm working on a Rosie RPL implementation in pure V, and rosie has *.rpl files. A bit like source code files. And I'm using the implementation already for some early use cases (rosies in-source-code unittests), requiring some parsing. That is, the RPL to match a rosie unittest, is in a file. The rpl is obviously static, and I'm using $embed_file only, because V has not multi-line string support.

**Comment by larpon** [comment]
> ... because V has not multi-line string support.

V has multi-line strings?

```v
ml_str := 'Line 1
Line 2
Line 3'
println(ml_str)
```
